### PR TITLE
Add Documentation Contribution Guide

### PR DESCRIPTION
**Pull request by ncbaratta**
### Type of change

- Enhancement / new feature

### Description

This adds a documentation contribution guide with guidelines on style and templates.

### CODE REVIEW

**Comment by ncbaratta** [comment]
@scholzj this is our contribution guide to use as a reference in training and in documenting going forward.

**Comment by scholzj** [comment]
@ncbaratta Is this ready to merge? Or should we wait for some more changes?

File: documentation/contributing/contributor-guide.adoc [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+. If the update is rejected because the commit is behind, merge your changes
++
+[source]
+----
+$ git pull upstream master
+$ git push -f origin HEAD
+----
+. Visit your fork on GitHub
+. Click *Compare & pull request*
+. If a SME review is needed:
```
Comment (tombentley) [inline documentation/contributing/contributor-guide.adoc]: Can we define SME, and how would a contributor know whether a SME review was needed?

File: documentation/contributing/formatting-guide.adoc [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+Monospace (```)::
+[cols="50%,50%",options="header"]
+|===
+|Item |Example
+|File names and paths |The \`/home/user/.bashrc` file ...
+|Literal values | If the value is \`true` ...
+|Configuration attributes | Set the \`enabled` attribute ...
+|Java class names |The \`java.sql.Connection` class ...
+|Java annotations |The \`@Clustered` annotation ...
```
Comment (tombentley) [inline documentation/contributing/formatting-guide.adoc]: I assume we should also use monospace for Kubernetes/OpenShift resource kinds (`Pod`, `ConfigMap` etc). Can we add that as an example so it's completely clear to contributors? Please could you also open an issue to convert the existing docs to using monospace for resource kinds?

File: documentation/contributing/contributor-guide.adoc (hunk 1 above)
Comment (ncbaratta) [reply documentation/contributing/contributor-guide.adoc]: Yes, let's enter an issue for me to add more info to this section.  To give you a quick answer - a SME Review is a Subject Matter Expert and a review by a SME is almost always needed. The only time you wouldn't need one is if you were fixing a broken link or a typo.

File: documentation/contributing/formatting-guide.adoc (hunk 2 above)
Comment (ncbaratta) [reply documentation/contributing/formatting-guide.adoc]: Let's add 2 new issues - 1 for a better example and 1 for fixing the existing docs.  @dmuntima and I are going through the existing docs to help convert them to templates and make edits like this, but would love some help!!!

**Comment by ncbaratta** [comment]
@scholzj it's ready for merging whenever you are!! I can make enhancements to it later.

File: documentation/contributing/formatting-guide.adoc (hunk 2 above)
Comment (scholzj) [reply documentation/contributing/formatting-guide.adoc]: I raised #609 and #608. Are GitHub issues enough for this? Or do we need JIRAs?

File: documentation/contributing/contributor-guide.adoc (hunk 1 above)
Comment (scholzj) [reply documentation/contributing/contributor-guide.adoc]: I raised #607. Are GitHub issues enough for this? Or do we need JIRAs?

**Comment by ncbaratta** [comment]
Thanks, @scholzj GitHub issues assigned to me are perfect!

**Pull request merged**